### ISSUE TITLE
Exporting a sample track cuts off the beginning

### ISSUE THREAD

**Issue by MisterTeeXD** [LMMS/lmms#4362]
LMMS Version: LMMS 1.2.0rc5, Win64
OS: Windows 10

Issue:
When a song gets exported, a small part of the beginning will get cut off

Steps To Reproduce: 
1: Go to File
2: Select Export
3: Select Save (Any Directory)
4: Select Export
5: Play the Audio File

This issue was also in LMMS 1.1.3 and it is also in LMMS 1.2.0rc5, the audio file, when fixed, will play without the cutoff of the beginning

**Comment by SecondFlight** [comment]
Could you please update this with a project file where this happens for you? I haven't noticed the same issue, though I could have just missed it. How much of the beginning does it cut off?

**Comment by musikBear** [comment]
Also what export-settings, and export file-type, if it matters

**Comment by PhysSong** [comment]
It always happens for me. The length of missing samples are one buffer (<= 256 samples) so it isn't easy to notice the issue.

**Comment by musikBear** [comment]
@PhysSong, can you make a demo project?

**Comment by PhysSong** [comment]
I've done more test, and here's the result:
- In 1.1, silence of a buffer is added.
- In 1.2, a part of sample is silenced(see the picture).
![missing](https://user-images.githubusercontent.com/12591074/40266306-e1eab6a4-5b83-11e8-9775-8d7fa5cae09b.png)

The issue in 1.1 have been fixed in #1370. However, #3133 have introduced a new issue. A call to `Song::startExport` emits`playbackStateChanged` signal, and connected `SampleTCO::playbackPositionChanged` slot will be called in **UI thread**. Due to the message queuing, the actual execution is done while exporting.
Unfortunately, using `Qt::DirectConnection` introduces a new deadlock related to `requestChangeInModel`. The function needs to be unblocked by mixer, but mixer won't work before the function is unblocked. It's something like #3344 though not very similar.

Summary: a bug in 1.1 was fixed, but a new bug was introduced between 1.2.0-rc1 and 1.2.0-rc2.

**Comment by SecondFlight** [comment]
@PhysSong Thanks for researching this!

**Comment by PhysSong** [comment]
I think we should use `Qt::DirectConnection` and fix the threading issues. I have an idea to fix this and will implement it soon. Since this bug affects exporting, I'll finish it before the 1.2 release.
